### PR TITLE
Hotkey for splitting cells

### PR DESCRIPTION
**Pull request by robmck1995**
Fixes: https://github.com/marimo-team/marimo/issues/1372

The issue mentioned using Ctrl+Shift+- as the hotkey, but that seemed to have some interaction with zooming in the browser.

I have the hotkey set to Mod+Shift+' for now; if you'd like to change that, I'm happy to do so!

Also, tested in the browser but could not make unit tests for the reducer due to the dependency on the editor view. I have added a TODO to find a solution to this.

Solved with the help of Glide. Check out the full solution here!
https://glide.agenticlabs.com/sharing?userId=65bae76b7fbfb1c20ac1f3b8&taskId=miIvJdg

### CODE REVIEW

**Comment by mscolnick** [comment]
Lets go Glide! 🤖

File: frontend/src/core/cells/cells.ts [hunk 1]
```diff
@@ -603,6 +607,70 @@ const {
...
     };
   },
+  splitCell: (state, action: { cellId: CellId; cursorPos: number }) => {
+    const { cellId, cursorPos } = action;
+    const index = state.cellIds.indexOf(cellId);
+    const cell = state.cellData[cellId];
+    const cellHandle = state.cellHandles[cellId].current;
+
+    if (cellHandle?.editorView == null) {
+      // TODO: Because of this we can't do  the reducer tests like for the other functions
```
Comment (mscolnick) [inline frontend/src/core/cells/cells.ts:617]: ill handle this todo in a follow up

File: frontend/src/core/codemirror/cells/extensions.ts [hunk 2]
```diff
@@ -236,6 +240,21 @@ export function cellMovementBundle(
         return true;
       },
     },
+    {
+      key: HOTKEYS.getHotkey("cell.splitCell").key,
+      preventDefault: true,
+      stopPropagation: true,
+      run: (ev) => {
+        const cursorPos = ev.state.selection.main.head;
+        splitCell({ cellId, cursorPos }),
```
Comment (mscolnick) [inline frontend/src/core/codemirror/cells/extensions.ts:249]: i think you want a `;` at the end of the line: `splitCell({ cellId, cursorPos });`

**Comment by robmck1995** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by robmck1995** [comment]
recheck

**Pull request merged**